lib/Models/Catalog/CatalogGroups/ThreddsCatalogGroup.ts

```diff
@@ -122,7 +122,7 @@ export class ThreddsStratum extends LoadableStratum(ThreddsCatalogGroupTraits) {
     // Create members for individual datasets
     for (let i = 0; i < this.threddsCatalog.datasets.length; i++) {
       const ds = this.threddsCatalog.datasets[i];
-      await ds.loadAllNestedCatalogs();
+      ds.loadAllNestedCatalogs();
```

Comment:
Actually the typing [here](https://github.com/TerriaJS/terriajs/blob/0831d21b0f1b69ebac301918b8af3dcc13afb745/lib/Models/Catalog/CatalogGroups/ThreddsCatalogGroup.ts#L32-L33) is incorrect. The return value must be `Promise<void>` instead of `void`

The actual implementation comes from[ this repo](https://github.com/TerriaJS/thredds-catalog-crawler/blob/802b112be02069c90034bd8ff66743e30c27a559/src/Catalog.js#L104) where you can see that these functions are defined as `async`. 

Same for the other [load method](https://github.com/TerriaJS/terriajs/blob/0831d21b0f1b69ebac301918b8af3dcc13afb745/lib/Models/Catalog/CatalogGroups/ThreddsCatalogGroup.ts#L46).

Reply:
That seems like a separate issue from what this PR was trying to tackle. I've dropped the Thredds change from this PR and will make a separate PR for that later today if I can find the time.

Reply:
It was easier than I thought, the Thredds types are fixed in #7356.